### DISCUSSION TITLE
uninstall and alternate install location

### DISCUSSION THREAD

**Discussion by t794104** [chocolatey/choco#2475] [Q&A] [answered]
I have a package that installs default to 'c:\program files\blabla' and i can override the locations with a installParam. Which works fine. With useRememberedArgumentsForUpgrades upgrade work also.

But now the problem; when i do a uninstall i need too delete some files in the uninstall.ps1 script. But i don't know where the install location is; the installParam used when installing the package is not available to the uninstall script (at least i can't find it).

So how do i get the (alternat) install location in the uninstall script ?

**Comment by TheCakeIsNaOH** [comment, marked as answer]
Assuming that the installer creates a registry key with uninstall information, I would suggest using `Get-UninstallRegistryKey` to find the install location. Uninstall keys may have a `InstallLocation` specified, or you may be able to get the directory from the `UninstallString` in some cases.

**Reply by t794104** [reply]
There is no installer. We are using Choco as installer for some config / xml files. So the install script copy's file to a install location. And with the uninstall i'd like too remove them. I just can't find them :)

**Reply by TheCakeIsNaOH** [reply]
Ok, in that case, create a file in the tools directory (e.g. next to the `chocolateyInstall.ps1` called something like `install-location.txt`, and put the path of the install location in there.

Then during uninstall, try to read that file, and try to uninstall from the location it specifies. If the file does not exist, then try to uninstall from the default location. If the default install location does not exist, output a warning that you cannot find the install location and it must be removed manually.

Another option could be to create the uninstall registry keys, which then could be read.  This would make the program show up in programs & features as a normal windows program does.
